notion-core/src/test/scala/com/ambiata/notion/core/LocationTemporary.scala

```diff
@@ -43,6 +43,16 @@ case class LocationTemporary(t: T, path: String, client: AmazonS3Client, conf: C
 }
 
 object LocationTemporary {
+
+  /** @return an arbitrary location where the path possibly ends with a / */
   implicit def LocationTemporaryArbitrary: Arbitrary[LocationTemporary] =
-    Arbitrary(arbitrary[T].map(LocationTemporary(_, java.util.UUID.randomUUID().toString, Clients.s3, new Configuration)))
+    Arbitrary {
+      val path = java.util.UUID.randomUUID.toString
+
+      arbitrary[T].flatMap { t =>
+        val locationTemporary = LocationTemporary(t, path, Clients.s3, new Configuration)
+        val locationDirTemporary = LocationTemporary(t, path + "/", Clients.s3, new Configuration)
```

Comment:
That says something about the notion of a "directory" under S3. Not the same thing as Unix. I think that we should return `isDirectory = true` and `isFile = true` in this case with an additional `isFile` method. And we should make clear that for an arbitrary `Location`: 
```
!forAll { location =>
   isDirectory(location) == !isFile(location)
}
```